src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -1341,6 +1341,51 @@ void IgnRenderer::XYZConstraint(math::Vector3d &_axis)
   }
 }
 
+/////////////////////////////////////////////////
+bool IgnRenderer::ContainsSimpleShape(const rendering::NodePtr &_node,
+    rendering::VisualPtr &_visual) const
+{
+  std::queue<rendering::NodePtr> q;
+
+  // Adding the root node.
+  q.push(_node);
+
+  while (!q.empty())
+  {
+    // Process the next node.
+    auto n = q.front();
+    q.pop();
+    if (n)
+    {
+      auto v = std::dynamic_pointer_cast<ignition::rendering::Visual>(n);
+      if (v)
+      {
+        try
+        {
+          int userData = std::get<int>(v->UserData("geometry-type"));
+
+          if (userData == static_cast<int>(sdf::GeometryType::BOX)      ||
+              userData == static_cast<int>(sdf::GeometryType::CYLINDER) ||
+              userData == static_cast<int>(sdf::GeometryType::SPHERE))
```

Comment:
Can you add a TODO to add ellipsoid when we forward-port to Edifice?

Scaling capsules is trickier, so I think we don't need  to do it in the first pass.

Reply:
Done in 7d0728a.

Reply:
expand todo note to include `sdf::GeometryType::ELLIPSOID`?

Reply:
Done in 3b857a9.